### PR TITLE
Move IncomingRequestAddressStrategy::determineServletContextPath into IServerAddressStrategy to permit overriding

### LINKED ISSUE #6038: Allow overriding of IncomingRequestAddressStrategy::determineServletContextPath
**Describe the issue**
I am working with a HAPI FHIR server that is fronted by a reverse proxy. The reverse proxy accepts traffic at mydomain.com/myapp/fhir/r4/QuestionnaireResponse and strips off the "/myapp" and send the resulting request to the server. It includes the not-officially-standard-but-widely-used `X-Forwarded-Prefix` header of "/myapp". This prefix can change and so we do not want want to require the application to be explicitly configured with a particular prefix.

We are using the spring `server.forward-headers-strategy=framework` property, which will correctly interpret the `X-Forwarded-Prefix` and reflect it appropriately in calls to `request.getContextPath()`. We need to get the HAPI FHIR RestfulServer to use that when parsing the incoming request. Unfortunately, https://github.com/hapifhir/hapi-fhir/blob/0397b9ddc8a8672d64fa726c26c13fbfc7b97a25/hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/RestfulServer.java#L1050 is hard coded to use the `IncomingRequestAddressStrategy.determineServletContextPath()` static method. This ends up using `server.getServletContext().getContextPath()`, which returns an empty string in my use case. Ultimately, this leads to the HAPI FHIR Server being unable to properly parse the incoming requests.

It would be great if this could be overridden.

**Environment (please complete the following information):**
 - HAPI FHIR Version: 7.0.2
 - OS: Linux
 - Browser N/A

### PR DESCRIPTION
**Pull request by alexrkopp**
Fixes #6038

### CODE REVIEW

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/IServerAddressStrategy.java [hunk 1]
```diff
@@ -32,4 +33,35 @@ public interface IServerAddressStrategy {
...
+
+	/**
+	 * Determines the servlet's context path.
+	 *
+	 * This is here to try and deal with the wide variation in servers and what they return.
+	 *
+	 * getServletContext().getContextPath() is supposed to return the path to the specific servlet we are deployed as but it's not available everywhere. On some servers getServletContext() can return
+	 * null (old Jetty seems to suffer from this, see hapi-fhir-base-test-mindeps-server) and on other old servers (Servlet 2.4) getServletContext().getContextPath() doesn't even exist.
+	 *
+	 * theRequest.getContextPath() returns the context for the specific incoming request. It should be available everywhere, but it's likely to be less predicable if there are multiple servlet mappings
```
Comment (blingenau) [inline hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/IServerAddressStrategy.java:45]:
```suggestion
	 * theRequest.getContextPath() returns the context for the specific incoming request. It should be available everywhere, but it's likely to be less predictable if there are multiple servlet mappings
```

**Comment by alexrkopp** [comment]
@jmarchionatto, I see you frequently do reviews/approvals. Is there a process I should be following to officially request a review of this change?

**Review by jmarchionatto — APPROVED** [review]
Looks good! Approved.

**Comment by alexrkopp** [comment]
Thanks @jmarchionatto. I'd like to make sure this change is on track for the next release. When can I expect this PR to be merged? Additionally, is there an established release cadence? I wasn't able to find any information on the public documentation/contribution guide.

**Comment by jmarchionatto** [comment]
You are welcome, @alexrkopp I'll look after this PR. Please merge master into for an updated build

**Comment by alexrkopp** [comment]
@jmarchionatto I do not have repository permission to actually do the merge. Likely because I am a third party contributor.

**Comment by jmarchionatto** [comment]
@alexrkopp I will do the merge into master for you, but we need a clean build before I could do that. What I meant is for you to merge master into your branch, so a build with updated code takes place.

**Comment by alexrkopp** [comment]
@tadgh thanks for reaching out. Yes. That is fine.

**Pull request closed without merging**